sql/src/main/java/io/crate/executor/transport/StreamBucket.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.executor.transport;
+
+import com.google.common.base.Function;
+import com.google.common.base.Throwables;
+import io.crate.Streamer;
+import io.crate.core.collections.Bucket;
+import io.crate.core.collections.Row;
+import io.crate.core.collections.RowN;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Iterator;
+
+public class StreamBucket implements Bucket, Streamable {
+
+    private Streamer<?>[] streamers;
+    private int size = -1;
+    private BytesReference bytes;
+
+    public static class Builder {
+        private int size = 0;
+        private final Streamer<?>[] streamers;
+        private final BytesStreamOutput out;
+
+        public static final Function<Builder, Bucket> BUILD_FUNCTION =
+                new Function<Builder, Bucket>() {
+                    @Nullable
+                    @Override
+                    public Bucket apply(StreamBucket.Builder input) {
+                        try {
+                            return input.build();
+                        } catch (IOException e) {
+                            Throwables.propagate(e);
+                        }
+                        return null;
+                    }
+                };
+
+        public Builder(Streamer<?>[] streamers) {
+            this.streamers = streamers;
+            this.out = new BytesStreamOutput();
```

Comment:
good catch. however i thought it is ok since it is per bucket, not per row as in other cases. however, will set this to 1k